builder/ebs/step_enable_fast_launch.go

```diff
@@ -0,0 +1,112 @@
+package ebs
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	awscommon "github.com/hashicorp/packer-plugin-amazon/builder/common"
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+	"github.com/hashicorp/packer-plugin-sdk/retry"
+)
+
+type stepEnableFastLaunch struct {
+	PollingConfig      *awscommon.AWSPollingConfig
+	AMISkipCreateImage bool
+	EnableFastLaunch   bool
+	MaxInstances       int
+	ResourceCount      int
+}
+
+func (s *stepEnableFastLaunch) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	if !s.EnableFastLaunch {
+		// Nothing to do if fast boot is disabled
+		return multistep.ActionContinue
+	}
+
+	ec2conn := state.Get("ec2").(*ec2.EC2)
+	ui := state.Get("ui").(packersdk.Ui)
+	amis := state.Get("amis").(map[string]string)
+
+	if s.AMISkipCreateImage {
+		ui.Say("Skipping fast-launch setup...")
+		return multistep.ActionContinue
+	}
```

Comment:
I believe this check has to happen before line 32, as "amis" is only added to state if a user does not specify the skip create image configuration. 

Alternatively, you could use the state.GetOk function to check if there is an amis map in State.

Reply:
You are completely right, good catch!